### PR TITLE
mobile error codes - auth bug

### PR DESCRIPTION
**Pull request by WesJourdan**
NOTE: to be merged into open PR #214 

- Update login and register methods on server to return user and token, rm unused mobileLogin method.
- Update mobile auth store to deal with server returned tokens.
- Remove login action from mobile register page (no longer necessary, user is logged in on register).
- Clean up a couple of small things I noticed.

### CODE REVIEW

File: server/resources/user/sessionController.js [hunk 1]
```diff
@@ -161,7 +108,22 @@ exports.register = async (req, res) => {
       if(req.ip && typeof(req.ip) == 'string' ) {
         req.session.ip = req.ip;
       }
-      res.json(safeUser);
+
+      // if this is a mobile request we need to return the token
+      if(req.body.mobile) {
+        console.log('MOBILE REQUEST BEING FULFILED LIKE A BAAAUUUUCE');
```
Comment (austincline09) [inline server/resources/user/sessionController.js]: lol

Comment (WesJourdan) [reply server/resources/user/sessionController.js]: lmao whoops!

Comment (WesJourdan) [reply server/resources/user/sessionController.js]: Fixed

File: mobile/Yote/js/AuthNavigator.js [hunk 2]
```diff
@@ -26,17 +26,17 @@ export default function AuthNavigator() {
...
             <AuthStack.Screen
               name="UserLogin"
               component={UserLogin}
               options={{
-                title: 'Sign in',
+                title: 'Sign in'
                 // When logging out, a pop animation feels intuitive
                 // You can remove this if you want the default 'push' animation
-                animationTypeForReplace: loggedInUser ? 'pop' : 'push',
+                , animationTypeForReplace: loggedInUser ? 'pop' : 'push'
```
Comment (grantfowler) [inline mobile/Yote/js/AuthNavigator.js:39]: +1

File: mobile/Yote/js/resources/user/authStore.js [hunk 3]
```diff
@@ -20,7 +22,9 @@ export const sendRegister = createAsyncThunk(
 export const sendLogin = createAsyncThunk(
   'auth/sendLogin'
   , async (userInfo) => {
-    const response = await apiUtils.callAPI('/api/users/mobile-login', 'POST', userInfo);
+    // let the server know this is a mobile request so the session token is returned with the user
+    userInfo.mobile = true;
```
Comment (grantfowler) [inline mobile/Yote/js/resources/user/authStore.js:26]: can you add this field to the register action too?

**Review by grantfowler — CHANGES_REQUESTED** [review]
small change

**Review by grantfowler — APPROVED** [review]

**Pull request merged**